vaadin-button.html

```diff
@@ -22,25 +22,178 @@
   <template>
     <style>
       :host {
-        display: block;
+        align-items: flex-start;
+        background: linear-gradient(to bottom right, #fff, #eee);
+        border: 1px solid #999;
+        border-radius: 2px;
+        box-sizing: border-box;
+        cursor: default;
+        display: inline-block;
+        font-size: 15px;
+        letter-spacing: normal;
+        margin: 0;
+        padding: 2px 8px;
+        text-align: center;
+        text-indent: 0;
+        text-rendering: auto;
+        text-shadow: none;
+        text-transform: none;
+        word-spacing: normal;
+        user-select: none;
+      }
+
+      :host(:hover) {
+        border-color: #666;
+      }
+
+      :host([disabled]) {
+        color: #aaa !important;
+        background: #eee !important;
+        border-color: #aaa !important;
+        pointer-events: none;
+      }
+
+      :host([active]) {
+        background: linear-gradient(to bottom right, #eee, #fff);
       }
     </style>
+    <slot id="label"></slot>
   </template>
 
   <script>
-    class VaadinElement extends Polymer.Element {
-      static get is() {
-        return 'vaadin-button';
-      }
+    HTMLImports.whenReady(function() {
+      class VaadinButton extends Polymer.Element {
+        static get is() {
+          return 'vaadin-button';
+        }
+
+        static get config() {
+          return {
+            properties: {
+              /**
+               * If true, the button is in the active state (clicked).
+               */
+              active: {
+                type: Boolean,
+                default: false,
+                reflectToAttribute: true
+              },
+
+              /**
+               * Specify that this control should have input focus when the page loads.
+               */
+              autofocus: {
+                type: Boolean
+              },
+
+              /**
+               * Indicates that this control is not available for interaction.
+               */
+              disabled: {
+                type: Boolean,
+                default: false,
+                reflectToAttribute: true,
+                observer: '_disabledChanged',
+              },
+
+              /**
+               * The position of the element in the tabbing navigation order for the current document.
+               */
+              tabindex: {
+                type: Number,
+                value: 0,
+                reflectToAttribute: true
+              },
+
+              /**
+               *
+               */
+              _previousTabIndex: Number
+            }
+          };
+        }
+
+        ready() {
+          this._addListeners();
+
+          super.ready();
+
+          if (this.autofocus) {
+            this.focus();
+          }
+        }
+
+        connectedCallback() {
+          super.connectedCallback();
+          this._lightDOMobserver = Polymer.dom(this).observeNodes(this._processLightDOM);
```

Comment:
Observing the light DOM is unnecessary. Given that there is only a single unnamed slot, all the light DOM content should distribute there automatically.